src/oaklib/interfaces/differ_interface.py

```diff
@@ -282,3 +305,345 @@ def compare_term_in_two_ontologies(
         self, other_ontology: BasicOntologyInterface, curie: CURIE, other_curie: CURIE = None
     ) -> Any:
         raise NotImplementedError
+
+    def diff_structured(
+        self,
+        other_ontology: BasicOntologyInterface,
+        configuration: DiffConfiguration = None,
+    ) -> Iterator[Dict[str, Any]]:
+        """
+        Provides a structured diff of two ontologies
+
+        Required sequence of changes:
+        1. New classes
+        2. Label changes
+        3. Definition changes
+        4. Obsoletions
+        5. Added synonyms
+        6. Added definitions
+        7. Changed relationships
+
+        :param other_ontology: Ontology to compare against
+        :param configuration: Configuration for the differentiation
+        :return: A sequence of changes in the form of a dictionary
+        """
+        if configuration is None:
+            configuration = DiffConfiguration()
+        other_ontology_entities = set(list(other_ontology.entities(filter_obsoletes=False)))
+        self_entities = set(list(self.entities(filter_obsoletes=False)))
+
+        # ! New classes
+        """
```

Comment:
this is odd, why doesn't the linter complain? keep docstrings together!

Reply:
It is not a docstring per se. Just a comment for me to remember in the future as to what the code block means.